### PR TITLE
cmd/openshift-install/create: Log progress on timeout too

### PR DESCRIPTION
**Pull request by wking**
Since fd1349ce31 (#1432), we log progress messages while waiting.  But I'd forgotten to log them in the timeout message, which could lead to logs like:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-4.0/5971/artifacts/e2e-aws/installer/.openshift_install.log | grep -B3 level=fatal
time="2019-03-21T02:17:31Z" level=debug msg="Still waiting for the cluster to initialize: Working towards 4.0.0-0.ci-2019-03-21-015242: 95% complete"
time="2019-03-21T02:36:11Z" level=debug msg="Still waiting for the cluster to initialize: Could not update rolebinding \"openshift-cluster-storage-operator/cluster-storage-operator\" (231 of 310): the server has forbidden updates to this resource"
time="2019-03-21T02:36:41Z" level=debug msg="Still waiting for the cluster to initialize: Working towards 4.0.0-0.ci-2019-03-21-015242: 98% complete"
time="2019-03-21T02:42:24Z" level=fatal msg="failed to initialize the cluster: timed out waiting for the condition"
```

That's not very helpful if you're looking at stderr and our default info level.  With this commit, we'll get:

```
failed to initialize the cluster: timed out waiting for the condition: Working towards 4.0.0-0.ci-2019-03-21-015242: 98% complete
```

Even better would be to get the "forbidden updates to this resource" message, but that's up to the cluster-verison operator to set more helpful failing/progressing messages.

### CODE REVIEW

**Comment by wking** [comment]
For kicks, here's the CVO waterfall showing the 25-minute hangs for a few components:

cvo

Generated with:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-4.0/5971/artifacts/e2e-aws/pods/openshift-cluster-version_cluster-version-operator-6d66c49cfd-tbtdz_cluster-version-operator.log.gz | gunzip | cvo-waterfall.py >cvo.svg
$ convert cvo.svg cvo.png  # boo, GitHub not liking inline SVG in comments ;)
```

Slow components:

```console
$ grep 'width="[0-9][0-9][0-9]' cvo.svg
  <rect x="0.103089" y="340" width="1491.3961180000001" height="4" fill="blue"><title>servicemonitor openshift-apiserver-operator/openshift-apiserver-operator 306/310 (0:24:51.396118)</title></rect>
  <rect x="0.104142" y="344" width="1490.148234" height="4" fill="blue"><title>servicemonitor openshift-kube-scheduler-operator/kube-scheduler-operator 303/310 (0:24:50.148234)</title></rect>
  <rect x="0.105381" y="360" width="1491.321917" height="4" fill="blue"><title>servicemonitor openshift-kube-apiserver-operator/kube-apiserver-operator 297/310 (0:24:51.321917)</title></rect>
  <rect x="0.16184" y="424" width="1490.714758" height="4" fill="blue"><title>servicemonitor openshift-controller-manager-operator/openshift-controller-manager-operator 309/310 (0:24:50.714758)</title></rect>
  <rect x="0.216951" y="432" width="1491.359591" height="4" fill="blue"><title>servicemonitor openshift-kube-controller-manager-operator/kube-controller-manager-operator 300/310 (0:24:51.359591)</title></rect>
  <rect x="1.679898" y="588" width="1491.5600559999998" height="4" fill="blue"><title>rolebinding openshift-cluster-storage-operator/cluster-storage-operator 231/310 (0:24:51.560056)</title></rect>
  <rect x="3.256621" y="736" width="104.000343" height="4" fill="blue"><title>clusteroperator openshift-controller-manager 146/310 (0:01:44.000343)</title></rect>
  <rect x="12.918625" y="920" width="1477.358464" height="4" fill="blue"><title>servicemonitor openshift-service-catalog-apiserver-operator/openshift-service-catalog-apiserver-operator 291/310 (0:24:37.358464)</title></rect>
  <rect x="13.200038" y="944" width="1478.902274" height="4" fill="blue"><title>servicemonitor openshift-image-registry/image-registry 288/310 (0:24:38.902274)</title></rect>
  <rect x="13.240177" y="952" width="1477.689671" height="4" fill="blue"><title>servicemonitor openshift-service-catalog-controller-manager-operator/openshift-service-catalog-controller-manager-operator 294/310 (0:24:37.689671)</title></rect>
  <rect x="13.899698" y="992" width="136.002139" height="4" fill="blue"><title>clusteroperator node-tuning 209/310 (0:02:16.002139)</title></rect>
  <rect x="14.621614" y="1048" width="149.000247" height="4" fill="blue"><title>clusteroperator image-registry 185/310 (0:02:29.000247)</title></rect>
  <rect x="15.122052" y="1072" width="149.000347" height="4" fill="blue"><title>clusteroperator ingress 195/310 (0:02:29.000347)</title></rect>
  <rect x="15.172068" y="1076" width="172.000617" height="4" fill="blue"><title>clusteroperator openshift-samples 221/310 (0:02:52.000617)</title></rect>
  <rect x="16.452695" y="1088" width="213.000302" height="4" fill="blue"><title>clusteroperator monitoring 201/310 (0:03:33.000302)</title></rect>
  <rect x="17.772878" y="1144" width="118.000463" height="4" fill="blue"><title>clusteroperator console 268/310 (0:01:58.000463)</title></rect>
```

More discussion of this phenomenon [here][1].

[1]: https://github.com/openshift/cluster-authentication-operator/issues/95#issuecomment-474512255

File: cmd/openshift-install/create.go [hunk 1]
```diff
@@ -344,17 +344,10 @@ func waitForInitializedCluster(ctx context.Context, config *rest.Config) error {
...
 				}
-				if cov1helpers.IsStatusConditionTrue(cv.Status.Conditions, configv1.OperatorFailing) {
-					logrus.Debugf("Still waiting for the cluster to initialize: %v",
-						cov1helpers.FindStatusCondition(cv.Status.Conditions, configv1.OperatorFailing).Message)
-					return false, nil
-				}
-				if cov1helpers.IsStatusConditionTrue(cv.Status.Conditions, configv1.OperatorProgressing) {
-					logrus.Debugf("Still waiting for the cluster to initialize: %v",
-						cov1helpers.FindStatusCondition(cv.Status.Conditions, configv1.OperatorProgressing).Message)
+				if status := clusterVersionString(cv); status != "" {
```
Comment (abhinavdahiya) [inline cmd/openshift-install/create.go]: why not drop the global `var clusterVersion *configv1.ClusterVersion` and switch status to `lastErr`, them you can do something like:
```
lastErr = nil
UntilWithSync
   available -> (true, nil)
   !available ->
     lastErr = <why failing or progressing>
     debug lastErr
     (false, nil)
err != nil && err == wait.ErrTimeout
  lastErr -> (lastErr)
  (err)
(nil)
```

Comment (wking) [reply cmd/openshift-install/create.go]: > why not drop the global `var clusterVersion *configv1.ClusterVersion` and switch status to `lastErr`...

The messages are just strings, so I've pushed b50248e4c -> 48a4d4d57 moving to a `var lastError string`.  How does that look?

**Comment by wking** [comment]
Testing with:

```diff
--- a/cmd/openshift-install/create.go
+++ b/cmd/openshift-install/create.go
@@ -321 +321 @@ func waitForInitializedCluster(ctx context.Context, config *rest.Config) error {
-	timeout := 30 * time.Minute
+	timeout := 4 * time.Minute
```

gave me:

```
INFO Waiting up to 4m0s for the cluster at https://api.wking.devcluster.openshift.com:6443 to initialize... 
FATAL failed to initialize the cluster: Working towards 4.0.0-0.alpha-2019-03-22-062941: 96% complete: timed out waiting for the condition 
```

I've pushed 48a4d4d57 -> 02a513678 to update the example in the commit message.

**Comment by wking** [comment]
All green :)

**Comment by wking** [comment]
And [in production][1]:

```
level=fatal msg="failed to initialize the cluster: Working towards 4.0.0-0.ci-2019-03-23-104434: 99% complete: timed out waiting for the condition"
```

:)

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-4.0/6098

**Pull request merged**